### PR TITLE
Issue 24 multilanguage support 

### PR DESCRIPTION
**Pull request by Florian-Sabonchi**
In this issue i added the localization in flutter for more information see https://docs.flutter.dev/development/accessibility-and-localization/internationalization and a function to change langauge at runtime. Currently language switching is not stored persistant because we need to disscuse for future how we store data persistant

### CODE REVIEW

**Comment by Florian-Sabonchi** [comment]
Closed PR #26 from @DasProffi to avoid merge conflics so we need to review bot commits

File: lib/main.dart [hunk 1]
```diff
@@ -13,12 +18,38 @@ class MyApp extends StatelessWidget {
...
+            localizationsDelegates: const [
+              AppLocalizations.delegate,
+              GlobalMaterialLocalizations.delegate,
+              GlobalWidgetsLocalizations.delegate,
+              GlobalCupertinoLocalizations.delegate,
+            ],
+            supportedLocales: const [
+              Locale('en', 'US'),
+              Locale('de', 'DE'),
+            ],
```
Comment (DasProffi) [inline lib/main.dart:47]: create a (const) tupel of languages with attributes which than is mapped to the locals or the strings needed for the setting_list_view e.g ("en","US","english") so changes only need to be made in one file

File: lib/pages/setting_list_view.dart [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:helpwave/services/language_model.dart';
+import 'package:provider/provider.dart';
+
+class SettingListView extends StatelessWidget {
+  const SettingListView({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    Map<String, String> languages = {"English": "en", "Deutsch": "de"};
```
Comment (DasProffi) [inline lib/pages/setting_list_view.dart:11]: see above

File: lib/pages/setting_list_view.dart [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:helpwave/services/language_model.dart';
+import 'package:provider/provider.dart';
+
+class SettingListView extends StatelessWidget {
```
Comment (DasProffi) [inline lib/pages/setting_list_view.dart:6]: change Name to "SettingLanguageSelctionPage" and file name to "SettingLanguageSelction"

**Review by DasProffi — CHANGES_REQUESTED** [review]

File: lib/main.dart [hunk 4]
```diff
@@ -1,24 +1,60 @@
...
+            localizationsDelegates: const [
+              AppLocalizations.delegate,
+              GlobalMaterialLocalizations.delegate,
+              GlobalWidgetsLocalizations.delegate,
+              GlobalCupertinoLocalizations.delegate,
+            ],
+            supportedLocales: [
+              for (var language in languages)
+                Locale(language["Shortname"]!, language["Local"])
+            ],
```
Comment (DasProffi) [inline lib/main.dart:52]: `languages.map((language) => Locale(language["Shortname"]!, language["Local"])).toList()`
has higher readability in my opinion

File: lib/pages/SettingLanguageSelection.dart [hunk 5]
```diff
@@ -0,0 +1,35 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:helpwave/main.dart';
+import 'package:helpwave/services/language_model.dart';
+import 'package:provider/provider.dart';
+
+class SettingLanguageSelection extends StatelessWidget {
```
Comment (DasProffi) [inline lib/pages/SettingLanguageSelection.dart:7]: should have a "Page" at the end for consistancy with other classes in pages

File: lib/main.dart [hunk 6]
```diff
@@ -1,24 +1,60 @@
...
 
 void main() {
   runApp(const MyApp());
 }
 
+List<Map<String, String>> languages = [
+  {"Name": "English", "Local": "US", "Shortname": "en"},
+  {"Name": "Deutsch", "Local": "DE", "Shortname": "de"}
+];
+
```
Comment (DasProffi) [inline lib/main.dart:19]: Unsure whether we want to leave this in the main.dart

Comment (Florian-Sabonchi) [reply lib/main.dart:19]: for the beginning i think it's okay

**Review by DasProffi — APPROVED** [review]

**Pull request merged**